Recruit-Training/Advanced-Recruit-Training/Embedded-Teaser/Training/Core/Src/Training.c

```diff
@@ -2,5 +2,130 @@
 
 void trainingTask(uint8_t* data)
 {
+    //Motor 1 data collecting
+    uint8_t Motor1forwardVsReverse = data[0] & 0b00000001;
+
+    uint8_t Motor1onOrOff = (data[0] & 0b10000000)>>7;
+
+    uint8_t Motor1Velocity = (data[0] & 0b00111110)>>1;
+
+    uint8_t Motor1VelocitySign = (data[0] & 0b01000000)>>6;
+
+    if (Motor1VelocitySign == 1)
+    {
+        Motor1Velocity = -Motor1Velocity;
+    } 
+
+
+    //Motor 2 Data Collecting
+    uint8_t Motor2forwardVsReverse = data[1] & 0b00000001;
+    
+    uint8_t Motor2onOrOff = (data[1] & 0b10000000)>>7;
+
+    uint8_t Motor2Velocity = (data[1] & 0b00111110)>>1;
+
+    uint8_t Motor2VelocitySign = (data[1] & 0b01000000)>>6;
+
+    if (Motor2VelocitySign == 1)
+    {
+        Motor2Velocity = -Motor2Velocity;
+    } 
+    
+    //Light Data Collecting
+    uint8_t LightInfoHeadLightsOff = data[2] & 0b00000001;
+
+    uint8_t LightInfoHeadLightsLow = (data[2] & 0b00000010)>>1;
+
+    uint8_t LightInfoHeadLightsHigh = (data[2] & 0b00000100)>>2;
+
+    uint8_t LightInfoRightSignal = (data[2] & 0b00001000)>>3;
+
+    uint8_t LightInfoLeftSignal = (data[2] & 0b00010000)>>4;
+
+    uint8_t LightInfoHazrad = (data[2] & 0b00100000)>>5;
+
+    uint8_t LightInfoBrakes = (data[2] & 0b01000000)>>6;
+
+
+    //what u have to do: prove that the car works, a bunch of if statements saying whether the car is working according to reality ✓
```

Comment:
Since this was a message for yourself you can delete it once you are done, it would only confuse future readers :P
If anything you could change it to say what you did and how you did as an overview of the file